src/runner.ts

```diff
@@ -0,0 +1,578 @@
+      }
+      fs.mkdirSync(workspaceDir);
+    }
+
+    // Sometimes a repo will be left in a bad state. Deleting it here
+    // will let it get cleaned up later.
+    for (let dir of fs.readdirSync(workspaceDir)) {
+      const repoDir = path.join(workspaceDir, dir);
+      if (!util.isDirSync(repoDir) || fs.readdirSync(repoDir).length === 1) {
+        if (this._verbose) {
+          console.log(`Removing clone ${repoDir}...`);
+        }
+        await promisify(rimraf)(repoDir);
+      }
+    }
+  }
+
+  /**
+   * @returns a dictionary object of dev dependencies from the bower.json
+   * entries in all workspace repos that are marked for test, suitable for
+   * serializing into the devDependencies key of a generated bower.json file
+   * for the workspace dir.
+   *
+   * TODO(usergenic): Merge strategy blindly overwrites previous value for key with whatever new value it encounters as we iterate through bower configs
+   * which may not be what we want.  Preserving the
+   * highest semver value is *probably* the desired approach
+   * instead.
+   */
+  _mergedTestRepoBowerConfig(): {
+    name: string; dependencies: {[key: string]: string};
+    resolutions: {[key: string]: string};
+  } {
+    const merged = {
+      name: 'generated-bower-config-for-tattoo-workspace',
+      dependencies: {},
+      resolutions: {}
+    };
+    for (const repo of Array.from(this._workspace.repos.values())
+             .filter(repo => repo.test)) {
+      const repoPath = path.join(this._workspace.dir, repo.dir);
+      // TODO(usergenic): Verify that we can assume bower.json is the config
+      // file in the event any repo-specific .bowerrc files are capable of
+      // redefining its name.
+      const bowerJsonPath = path.join(repoPath, 'bower.json');
+      if (!util.existsSync(bowerJsonPath)) {
+        continue;
+      }
+      let bowerJson = fs.readFileSync(bowerJsonPath).toString();
+      let bowerConfig = JSON.parse(bowerJson);
+      if (bowerConfig.devDependencies) {
+        for (const name in bowerConfig.devDependencies) {
+          merged.dependencies[name] = bowerConfig.devDependencies[name];
+        }
+      }
+      if (bowerConfig.dependencies) {
+        for (const name in bowerConfig.dependencies) {
+          merged.dependencies[name] = bowerConfig.dependencies[name];
+        }
+      }
+      if (bowerConfig.resolutions) {
+        for (const name in bowerConfig.resolutions) {
+          merged.resolutions[name] = bowerConfig.resolutions[name];
+        }
+      }
+    }
+    return merged;
+  }
+
+  /**
+   * Creates a .bowerrc that tells bower to use the workspace dir (`.`) as
+   * the installation dir (instead of default (`./bower_components`) dir.
+   * Creates a bower.json which sets all the workspace repos as dependencies
+   * and
+   * also includes the devDependencies from all workspace repos under test.
+   */
+  _installWorkspaceDependencies() {
```

Comment:
Just talked about this in person.  I think there could definitely be an opportunity to rely on bower for the git cloning feature; Could be great to reduce our dependency on nodegit which has some expensive binary install too (at least for node 6 I noticed).  We can utilize the GitHub API still for wildcard refs but otherwise, if experimentation shows bower install can achieve the rest of the git manipulation, would make sense to unify under that.  Worth a spike.